common/models/transducer/topology.py

```diff
@@ -0,0 +1,26 @@
+from .loss import rnnt_loss, rnnt_loss_out_type, rnnt_tf_loss, rna_tf_loss, rna_loss_out_type
+from .alignment import rnnt_alignment, rnnt_alignment_out_type, rna_alignment, rna_alignment_out_type
+
+
+class Topology:
+  def __init__(self, loss, loss_out_type, alignment, alignment_out_type, name):
```

Comment:
The types should be specified.

Reply:
beside name they are all functions.

Is something like this fine? Or should I add a base class for losses and out_types?
``` python
class Topology:
  def __init__(self, loss: function, loss_out_type: function, alignment: function,
               alignment_out_type: function, name: AnyStr):
    self.loss = loss
    self.loss_out_type = loss_out_type
    self.alignment = alignment
    self.alignment_out_type = alignment_out_type
    self.name = name
```

Reply:
> Is something like this fine?
> `loss: function` ...

No. Be more specific. See how this is documented in RETURNN. You could use `Callable`. But I think only to some degree. But then specify it exactly in the docstring.

> Or should I add a base class for losses and out_types?

No. This is a function. There is no base class.

> `name: AnyStr` ...

Also use `str` for `name`, not `AnyStr`. Be specific when possible.